flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/internals/KafkaTopicPartitionTest.java

```diff
@@ -26,14 +27,14 @@
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.fail;
 
-/** Tests for the {@link KafkaTopicPartition}. */
+/** Tests for the {@link TopicPartition}. */
 public class KafkaTopicPartitionTest {
 
     @Test
     public void validateUid() {
```

Comment:
You can remove this whole class. With Flink 2.0, we will not be able to restore state from old consumers anyways.

We should document (and I think this applies to Flink 2.0 more than to Kafka 4.0) that users need to migrate from SourceFunction/SinkFunction to Source/Sink with Flink 1.20 before switching to Flink 2.0.

Reply:
Note that the test at it's current state is absolutely meaningless.

Reply:
Hi, @AHeise.

At the beginning, for the convenience of review, I split the entire work of bump 2.0 into two PRs: [deleting deprecated APIs](https://github.com/apache/flink-connector-kafka/pull/139) and [adapting 2.0 APIs](https://github.com/apache/flink-connector-kafka/pull/140). 

However, this seems to have resulted in duplicate review work. I have already deleted the classes mentioned here in the [adapting 2.0 API](https://github.com/apache/flink-connector-kafka/pull/140). 

I plan to close this PR and let's trace it in a unified PR.